tests/operator_test.go

```diff
@@ -724,35 +805,55 @@ var _ = Describe("ALL Operator tests", func() {
-					resp := tests.MakePrometheusHTTPRequest(f, "query?query=kubevirt_cdi_cr_ready")
-					defer resp.Body.Close()
-					bodyBytes, err := ioutil.ReadAll(resp.Body)
-					if err != nil {
-						return false
-					}
-					err = json.Unmarshal(bodyBytes, &result)
-					if err != nil {
-						return false
-					}
-					if len(result["data"].(map[string]interface{})["result"].([]interface{})) == 0 {
-						return false
-					}
-					values := result["data"].(map[string]interface{})["result"].([]interface{})[0].(map[string]interface{})["value"].([]interface{})
-					for _, v := range values {
-						if s, ok := v.(string); ok {
-							if s == "1" {
-								return true
-							}
-						}
-					}
-					return false
-				}, 1*time.Minute, 1*time.Second).Should(BeTrue())
+				Eventually(func() int {
+					return getMetricValue("kubevirt_cdi_cr_ready")
+				}, 1*time.Minute, 1*time.Second).Should(BeNumerically("==", 1))
+			})
+
+			It("StorageProfile incomplete metric expected value when creating an incomplete profile", func() {
+				if !tests.IsPrometheusAvailable(f.ExtClient) {
+					Skip("This test depends on prometheus infra being available")
+				}
+
+				numAddedStorageClasses := 2
+				defaultStorageClass := utils.GetDefaultStorageClass(f.K8sClient)
+				defaultStorageClassProfile := &cdiv1.StorageProfile{}
+				err := f.CrClient.Get(context.TODO(), types.NamespacedName{Name: defaultStorageClass.Name}, defaultStorageClassProfile)
+				Expect(err).ToNot(HaveOccurred())
+				for i := 0; i < numAddedStorageClasses; i++ {
+					_, err = f.K8sClient.StorageV1().StorageClasses().Create(context.TODO(), createUnknownStorageClass(fmt.Sprintf("unknown-sc-%d", i)), metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+				}
+				originalMetricVal := getMetricValue("kubevirt_cdi_incomplete_storageprofiles_total")
+				expectedIncomplete := originalMetricVal + numAddedStorageClasses
+
+				Eventually(func() int {
+					return getMetricValue("kubevirt_cdi_incomplete_storageprofiles_total")
+				}, 2*time.Minute, 1*time.Second).Should(BeNumerically("==", expectedIncomplete))
+
+				By("Fix profiles to be complete and test metric value equals original")
+				for i := 0; i < numAddedStorageClasses; i++ {
+					profile := &cdiv1.StorageProfile{}
+					err = f.CrClient.Get(context.TODO(), types.NamespacedName{Name: fmt.Sprintf("unknown-sc-%d", i)}, profile)
+					Expect(err).ToNot(HaveOccurred())
+					// These might be wrong values, but at least the profile is no longer incomplete
+					profile.Spec.ClaimPropertySets = defaultStorageClassProfile.Status.ClaimPropertySets
+					err = f.CrClient.Update(context.TODO(), profile)
+					Expect(err).ToNot(HaveOccurred())
+					expectedIncomplete--
+					Eventually(func() int {
+						return getMetricValue("kubevirt_cdi_incomplete_storageprofiles_total")
+					}, 2*time.Minute, 1*time.Second).Should(BeNumerically("==", expectedIncomplete))
+				}
+				By("Delete unknown storage classes")
```

Comment:
You will want to put this in an AfterEach, because if there is test failure between creating and deleting, the deletion section will not get executed. And we want to make sure we clean up. AfterEach is executed even if the test failed.

Reply:
Done